internal/builders/istio.go

```diff
@@ -139,10 +140,25 @@ func (rf *FromBuilder) Get() *v1beta1.Rule_From {
 	return rf.value
 }
 
-// WithForcedJWTAuthorization adds RequestPrincipals = "*" requirement, forcing requests to use JWT authorization
-func (rf *FromBuilder) WithForcedJWTAuthorization() *FromBuilder {
+// WithForcedJWTAuthorization adds RequestPrincipals = "ISSUER/*" for every issuer, forcing requests to use JWT authorization
+func (rf *FromBuilder) WithForcedJWTAuthorization(accessStrategies []*gatewayv1beta1.Authenticator) *FromBuilder {
 	// Only support one source at the moment
-	source := v1beta1.Source{RequestPrincipals: []string{"*"}}
+	var requestPrincipals []string
+	for _, strategy := range accessStrategies {
+		if strategy.Name == "jwt" {
+			authentications := &Authentications{
+				Authentications: []*Authentication{},
+			}
+			if strategy.Config != nil {
+				_ = json.Unmarshal(strategy.Config.Raw, authentications)
```

Comment:
please do not ignore errors when unmarshalling Raw content. did you test if there is `Authorizations` in config as well? if not please add a unit test for it.

Reply:
It's the same way of getting that issuer as in:
https://github.com/kyma-project/api-gateway/blob/main/internal/builders/istio.go#L332
We're not checking for error there either.

Added `Authorizations` checking in unit test.

Reply:
we should fix both then? this is hiding a potential error, which relies solely on validation and we can not be certain all cases are checked. having error in such places will reduce time when investigating issues.

Reply:
seems like `json.Umarshal` on validate is enough to caught issues on Raw config level.